### PR TITLE
xattr support, fixes in consumer, using O_DIRECT to read prop files

### PR DESCRIPTION
**Pull request by deztructor**
Also , out-of-source build support and reusing cor macros

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,21 +1,35 @@
...
-  IF(CMAKE_SIZEOF_VOID_P EQUAL 4)
-    SET(LIB_SUFFIX "")
-  ELSE(CMAKE_SIZEOF_VOID_P EQUAL 4)
-    SET(LIB_SUFFIX 64)
-  ENDIF(CMAKE_SIZEOF_VOID_P EQUAL 4)
-ELSE()
-  SET(LIB_SUFFIX "")
-ENDIF()
+option(ENABLE_USER_SESSION "Enable user session support" OFF)
+option(ENABLE_ONESHOT "Is oneshot can be used" OFF)
```
Comment (thp) [inline CMakeLists.txt]: `s/Is/If/` ?

File: metafuse/include/metafuse.hpp [hunk 2]
```diff
@@ -864,22 +946,36 @@ class FuseFs
...
             update_uid();
             update_gid();
-            //argv_vec.push_back("-s");
-            argv_vec.push_back("-o");
-            argv_vec.push_back("default_permissions");
-            argv_vec.push_back("-o");
-            argv_vec.push_back(_uid.c_str());
-            argv_vec.push_back("-o");
-            argv_vec.push_back(_gid.c_str());
+            options["uid"] = std::to_string(::getuid());
```
Comment (thp) [inline metafuse/include/metafuse.hpp]: This change drops `default_permissions` from the options list - is that intended?

**Comment by thp** [comment]
Rest LGTM

File: metafuse/include/metafuse.hpp (hunk 2 above)
Comment (deztructor) [inline metafuse/include/metafuse.hpp]: yes, it is bettery if it is provided explicitely: default_permissions is not compatible with some other options on some fuse versions

File: CMakeLists.txt (hunk 1 above)
Comment (deztructor) [inline CMakeLists.txt]: maybe it sound clearer

**Pull request merged**